app/controllers/products_controller.rb

```diff
@@ -61,6 +61,20 @@ def destroy
     end
   end
 
+  # Uses a bit of metadata to check when the product was last
+  # updated, along with a hashed value called an Etag.
+  # Server will respond with an empty body if the content
+  # hasn't been updated.
+  def who_bought
+    @product = Product.find(params[:id])
+    @latest_order = @product.orders.order(:updated_at).last
+    if stale?(@latest_order)
+      respond_to do |format|
+        format.atom
```

Comment:
I know this is like this in the book so no worries, but a ruby best practice is to use an inline block if you're only sending one method to the parameter. It's also okay to use single letters when using inline instead of full word :) 

```suggestion
      respond_to do { |f| f.atom }
```